hibernate-core/src/test/java/org/hibernate/test/hql/HQLTest.java

```diff
@@ -1538,4 +1501,11 @@ public void testComponentNoAlias() throws Exception {
 		compileWithAstQueryTranslator( "from Human where name.first = 'Gavin'", false);
 	}
 
+	@Test
+	public void testCountDistinctOfSubquery(){
+		createNewQueryTranslator(
+				"select count (DISTINCT (SELECT function('lower', 'DAY') from Bar)) from Bar"
```

Comment:
You can use COUNT OVER() with window functions, but counting over a subselect from the SELECT clause? I don't think I ever saw this in the standard.